fix(telephony): disable some features when nic has tech role only

julieni opened the pull request:
Disable some features when user has only tech role

NB: eslint changed some foreign code, hope it's OK

antleblanc commented:
Hi @julieni!

We recently merged #1336 which apply Prettier rules so could you please rebase
your branch on `develop`?

To easily resolve conflicts you can **ignore** `HEAD` changes and only keep yours.

After that you can run:

```sh
$ yarn run lint
```

This command will lint the entire codebase.

or:

Amend your commit to trigger the `pre-commit` hook (prettier will be launched
under the hood).

Thanks!

julieni commented:
Hi @antleblanc,

I rebased my work and added some more changes about disabling features on lines.

I think there should be an improvement on the API side to be able to fecth admin/tech/billing info for all groups in one request

julieni commented:
Hi, any news on this PR ? I need these changes to let customers manage their services through the manager V6, as the "white label manager" is so buggy. Thanks.

FredericEspiau commented on packages/manager/apps/telecom/src/components/telecom/telephony/telephony-voip-service.service.js:
```diff
@@ -30,6 +33,19 @@ angular
               const telephonyGroup = new TelephonyGroup(item.value);
               groups[telephonyGroup.billingAccount] = telephonyGroup;
+
+              OvhApiTelephony.v6()
```
```suggestion
              return OvhApiMe.v6().get()
                  .then((me) => OvhApiTelephony.v6()
```

julieni replied:
Hi @FredericEspiau, I'm not sure about this one, we are in a forEach loop, so unless OvhApiMe caches results, we're going to have lots of useless requests.
Maybe I should chain OvhApiMe result with the fetch of billing accounts, will try that.

julieni replied:
> Maybe I should chain OvhApiMe result with the fetch of billing accounts, will try that.

done

FredericEspiau commented on packages/manager/apps/telecom/src/app/telecom/telephony/billingAccount/administration/telecom-telephony-billing-account-administration.controller.js:
```diff
@@ -2,7 +2,11 @@ angular
-    function TelecomTelephonyBillingAccountAdministrationCtrl($translate) {
+    function TelecomTelephonyBillingAccountAdministrationCtrl(
+      $translate,
+      $stateParams,
```
Could you please order alphabetically ? :)

julieni replied:
done

FredericEspiau commented on packages/manager/apps/telecom/src/app/telecom/telephony/billingAccount/administration/telecom-telephony-billing-account-administration.controller.js:
```diff
@@ -12,45 +16,52 @@ angular
-            text: $translate.instant(
-              'telephony_group_admin_actions_group_billing_options',
-            ),
+        TelephonyMediator.getGroup($stateParams.billingAccount).then(
```
```suggestion
        return TelephonyMediator.getGroup($stateParams.billingAccount).then(
```

all `Promise` should be returned so they can be chained :)

julieni replied:
done too ;)

antleblanc commented:
Thanks @julieni for your contributions and sorry about the delay of my response.

Theses permissions shouldn't be handled directly on the front part.
It relates to an API change that can tell if actions are allowed or not
regarding the nichandle role.

Currently the API allow users to access to these sections and allow you to
perform actions that's why implementing your changes aren't relevant.

But we notice your needs and we will put efforts in that way to make sure that
the user experience stay as smooth as possible.

Thank you!

julieni commented:
@antleblanc @FredericEspiau  : I know it's not a perfect fix as I can't contribute on the API itself (it would be a pleasure), but at least it does the job for non expert (without API knowledge) user using the manager as a tech only.

Right now it's not possible to give such an access to the customer to the V6, and the "white label" is buggy and incomplete too. So what's the solution to solve this quickly ?

I may suppose implementing rights check in your API calls should be done asap, considering it can be considered as security issue, and users may not even know this.

Plus, if this is a "wontfix", what was the point to wait 2 months and ask me to rebase and make cosmetic changes ? Seems just a waste of time right now...

antleblanc commented:
> Right now it's not possible to give such an access to the customer to the V6,
> and the "white label" is buggy and incomplete too. So what's the solution to
> solve this quickly ?

Thanks for reporting us this level of feedbacks regarding this issue but
currently we haven't any workaround.

Obviously the "White Label" interface has some lacks but we initiated some
discussions about it in order to provide a more sustainable solution.

Unfortunately I don't have any estimated date of arrival to share.

> I may suppose implementing rights check in your API calls should be done asap,
> considering it can be considered as security issue, and users may not even know
> this.

You are right! We will contact the API Team regarding your comments and we will
keep you posted.

> Plus, if this is a "wontfix", what was the point to wait 2 months and ask me to
> rebase and make cosmetic changes ? Seems just a waste of time right now...

Apologies for the back of forth within your Pull Request.

At first we wanted to integrate it as quickly as possible but we were warned
about the Issue I mentioned in my previous comment, that's why I decided to put
it as `wontfix`.

julieni commented:
Thanks for the feedback. We should keep the PR on hold, I think the code is OK, as fixes are now to be develop on the API side only.
On the performance side, please provide an endpoint to retrieve admin/tech/billing contacts on all groups in one request, as previously mentioned.

One last think btw, I think it would still be better than today to apply the PR, even if it's only a "cache misère" on the UI side. Because for customers using services in web or dedicated, giving tech access does not (I hope) allow to access billing, order or, even worse, terminate service. So I'm sure some people give tech access to VOIP services without knowing what it implies. Fixing the UI is a quick way to prevent mistakes from regular customers. It would not prevent hacks BTW, I totally agree on that.

julieni commented:
Hi @antleblanc, any news on this ? I still think that it would be far better than the current situation, and we can't wait for months or years hoping someone will put acl rules on 3 api endpoints one day...
